### PR TITLE
feature/header

### LINKED ISSUE #4: header/menu

### PR DESCRIPTION
**Pull request by mikepetrusha**
1. Add Side menu and User menu
2. Add private and public routes

### CODE REVIEW

File: src/components/Header/Header.tsx [hunk 1]
```diff
@@ -1,17 +1,43 @@
+import { useReactiveVar } from '@apollo/client';
+import MenuIcon from '@mui/icons-material/Menu';
+import { IconButton } from '@mui/material';
 import AppBar from '@mui/material/AppBar';
```
Comment (INeizvestnaya) [inline src/components/Header/Header.tsx]: one import is named and another is default

File: src/components/SideMenu/SideMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+import BadgeIcon from '@mui/icons-material/Badge';
+import CloseIcon from '@mui/icons-material/Close';
+import ContactPageIcon from '@mui/icons-material/ContactPage';
+import GroupIcon from '@mui/icons-material/Group';
+import GroupsIcon from '@mui/icons-material/Groups';
+import PsychologyIcon from '@mui/icons-material/Psychology';
+import TranslateIcon from '@mui/icons-material/Translate';
+import { Divider, MenuItem, MenuList } from '@mui/material';
+import Drawer from '@mui/material/Drawer';
+import IconButton from '@mui/material/IconButton';
```
Comment (INeizvestnaya) [inline src/components/SideMenu/SideMenu.tsx]: also some default imports

File: src/components/SideMenu/SideMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+          <MenuItem
+            onClick={() => {
+              navigate(RoutesPath.LANGUAGES);
+              onClose();
+            }}
+          >
+            <TranslateIcon sx={IconStyles} />
+            Languages
+          </MenuItem>
```
Comment (INeizvestnaya) [inline src/components/SideMenu/SideMenu.tsx]: Please, try to put away code duplication here with MenuItem components here. You can use array and configure onClick property to take path as an argument

File: src/components/UserMenu/UserMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
+import { useReactiveVar } from '@apollo/client';
+import AccountCircleIcon from '@mui/icons-material/AccountCircle';
+import Logout from '@mui/icons-material/Logout';
+import SettingsIcon from '@mui/icons-material/Settings';
+import { Avatar, Typography } from '@mui/material';
+import Divider from '@mui/material/Divider';
+import IconButton from '@mui/material/IconButton';
+import Menu from '@mui/material/Menu';
+import MenuItem from '@mui/material/MenuItem';
```
Comment (INeizvestnaya) [inline src/components/UserMenu/UserMenu.tsx]: also default imports

File: src/components/UserMenu/UserMenu.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const handleLogout = () => {
+    authService.clearStorage();
+    navigate(RoutesPath.LOGIN);
```
Comment (INeizvestnaya) [inline src/components/UserMenu/UserMenu.tsx]: with current router logic you don't need to navigate) Public and private routes redirect you automatically when auth status changes

File: src/route/AppRoute.tsx [hunk 6]
```diff
@@ -1,19 +1,119 @@
...
+          />
+          <Route
+            path={RoutesPath.SKILLS}
+            element={
+              <PrivateRoute>
+                <SkillsPage />
+              </PrivateRoute>
+            }
+          />
+
```
Comment (INeizvestnaya) [inline src/route/AppRoute.tsx:113]: maybe, also use an array here to avoid duplication

File: src/components/AuthHeader/AuthHeader.tsx [hunk 7]
```diff
@@ -14,21 +16,21 @@ export const AuthHeader = () => {
...
             color: 'white',
             minWidth: 150,
             fontWeight: 600
           }}
-          value="/signup"
+          value={RoutesPath.SIGNUP}
           component={NavLink}
           label="signup"
-          to="/signup"
+          to={RoutesPath.SIGNUP}
```
Comment (INeizvestnaya) [inline src/components/AuthHeader/AuthHeader.tsx:36]: I also noticed paths without constants in LoginPage and SignUpPage files

**Review by INeizvestnaya — COMMENTED** [review]
added comments in the code

File: src/components/Header/Header.tsx (hunk 1 above)
Comment (mikepetrusha) [reply src/components/Header/Header.tsx]: fixed

File: src/components/SideMenu/SideMenu.tsx (hunk 2 above)
Comment (mikepetrusha) [reply src/components/SideMenu/SideMenu.tsx]: fixed

File: src/components/SideMenu/SideMenu.tsx (hunk 3 above)
Comment (mikepetrusha) [reply src/components/SideMenu/SideMenu.tsx]: fixed

File: src/components/UserMenu/UserMenu.tsx (hunk 4 above)
Comment (mikepetrusha) [reply src/components/UserMenu/UserMenu.tsx]: fixed

File: src/components/UserMenu/UserMenu.tsx (hunk 5 above)
Comment (mikepetrusha) [reply src/components/UserMenu/UserMenu.tsx]: fixed

File: src/components/AuthHeader/AuthHeader.tsx (hunk 7 above)
Comment (mikepetrusha) [reply src/components/AuthHeader/AuthHeader.tsx:36]: fixed

**Review by INeizvestnaya — APPROVED** [review]

**Review by INeizvestnaya — APPROVED** [review]

**Pull request merged**